activestorage/test/models/attached/many_test.rb

```diff
@@ -916,20 +916,28 @@ def self.name; superclass.name; end
     assert_match(/Cannot find variant :unknown for User#highlights_with_variants/, error.message)
   end
 
-  test "transforms variants later" do
-    blob = create_blob(filename: "funky.jpg")
+  test "transforms variants later: enqueues when blob is representable" do
+    representable_blob = create_file_blob(filename: "racecar.jpg")
 
-    assert_enqueued_with job: ActiveStorage::TransformJob, args: [blob, resize_to_limit: [1, 1]] do
-      @user.highlights_with_preprocessed.attach blob
+    assert_enqueued_with job: ActiveStorage::TransformJob, args: [representable_blob, resize_to_limit: [1, 1]] do
+      @user.highlights_with_preprocessed.attach representable_blob
+    end
+  end
+
+  test "transforms variants later: does not enqueue when blob is unrepresentable" do
+    unrepresentable_blob = create_blob(filename: "hello.txt")
+
+    assert_no_enqueued_jobs only: ActiveStorage::TransformJob do
+      @user.highlights_with_preprocessed.attach unrepresentable_blob
     end
   end
 
   test "transforms variants later conditionally via proc" do
     assert_no_enqueued_jobs only: ActiveStorage::TransformJob do
-      @user.highlights_with_conditional_preprocessed.attach create_blob(filename: "funky.jpg")
+      @user.highlights_with_conditional_preprocessed.attach create_file_blob(filename: "racecar.jpg")
```

Comment:
I had to update tests here because : 
- they were failing because of:
  - `create_blob` creates a `text/plain` file by default. whereas there is already a helper `create_file_blob` that creates a representable with the proper content type

Reply:
which makes them "unrepresentable"